### PR TITLE
New simple plugin exercising the logging APIs

### PR DESCRIPTION
**Pull request by zwoop**

### CODE REVIEW

File: tests/tools/plugins/test_logging.cc [hunk 1]
```diff
@@ -0,0 +1,104 @@
+/** @file
+
+    ATS plugin to do (simple) regular expression remap rules
```
Comment (calavera) [inline tests/tools/plugins/test_logging.cc]: this is not quite correct 😆

**Comment by bryancall** [comment]
[approve ci autest]

**Comment by bryancall** [comment]
[approve ci autest]

File: tests/tools/plugins/test_logging.cc [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+static const char *PLUGIN_NAME = "test_logging";
+
+///////////////////////////////////////////////////////////////////////////////
+// Initialize the plugin.
+//
+TSReturnCode
+TSRemapInit(TSRemapInterface *api_info, char *errbuf, int errbuf_size)
+{
+  if (!api_info) {
+    strncpy(errbuf, "[tsremap_init] - Invalid TSRemapInterface argument", errbuf_size - 1);
```
Comment (SolidWallOfCode) [inline tests/tools/plugins/test_logging.cc]: Should use `snprintf` and `[%s] ...), PLUGIN_NAME`.

File: tests/tools/plugins/test_logging.cc [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  return TS_SUCCESS;
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// We don't have any specific "instances" here, at least not yet.
+//
+TSReturnCode
+TSRemapNewInstance(int argc, char *argv[], void **ih, char * /* errbuf ATS_UNUSED */, int /* errbuf_sizeATS_UNUSED */)
+{
+  char *filename = argv[2]; // Only one argument.
```
Comment (SolidWallOfCode) [inline tests/tools/plugins/test_logging.cc]: Check the arg count, to be safe.

**Comment by zwoop** [comment]
@SolidWallOfCode Yeh, that was copied from an existing plugin, but I'll fix it.

**Pull request closed without merging**